torch_xla/csrc/ir.h

```diff
@@ -113,15 +113,21 @@ class XlaNode : public torch::lazy::Node {
   torch::lazy::hash_t hash() const override { return dag_hash_; }
 
   torch::lazy::hash_t shapeHash() const override { return dag_hash_; }
+
+  torch::lazy::hash_t shardingHash() const { return sharding_hash_; }
+
   // The node's outputs get assigned the same HLO sharding
   // TODO: test multi-output example.
   const std::shared_ptr<xla::OpSharding> GetSharding() const {
     return output_sharding_;
   }
-  void SetSharding(const xla::OpSharding& sharding) {
-    output_sharding_ = std::make_shared<xla::OpSharding>(sharding);
+
+  void SetSharding(const xla::OpSharding& sharding);
+
+  void ClearSharding() {
+    output_sharding_ = nullptr;
+    sharding_hash_ = 0;
```

Comment:
I felt like we should not provide `ClearSharding` and `SetSharding`, we need to create new IR node if we want to modify the sharding.